plugins/modules/ecs_taskdefinition.py

```diff
@@ -46,9 +46,425 @@
     containers:
         description:
             - A list of containers definitions.
-        required: False
+            - See U(https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html) for a complete list of parameters.
+        required: True
         type: list
         elements: dict
+        suboptions:
+            name:
+                description: The name of a container.
+                required: False
+                type: str
+            image:
+                description: The image used to start a container.
+                required: False
+                type: str
+            repositoryCredentials:
+                description: The private repository authentication credentials to use.
+                required: False
+                type: dict
+                suboptions:
+                    credentialsParameter:
+                        description:
+                            - The Amazon Resource Name (ARN) of the secret containing the private repository credentials.
+                        required: True
+                        type: str
+            cpu:
+                description: The number of cpu units reserved for the container.
+                required: False
+                type: int
+            memory:
+                description: The amount (in MiB) of memory to present to the container.
+                required: False
+                type: int
+            memoryReservation:
+                description: The soft limit (in MiB) of memory to reserve for the container.
+                required: False
+                type: int
+            links:
+                description:
+                    - Allows containers to communicate with each other without the need for port mappings.
+                    - This parameter is only supported if I(network_mode=bridge).
+                required: False
+                type: list
+            portMappings:
+                description: The list of port mappings for the container.
+                required: False
+                type: list
+                elements: dict
+                suboptions:
+                    containerPort:
+                        description: The port number on the container that is bound to the user-specified or automatically assigned host port.
+                        required: False
+                        type: int
+                    hostPort:
+                        description: The port number on the container instance to reserve for your container.
+                        required: False
+                        type: int
+                    protocol:
+                        description: The protocol used for the port mapping. Valid values are tcp and udp.
```

Comment:
```suggestion
                        description: The protocol used for the port mapping.
```

Listing valid values is redundant here, there's choices

Reply:
Removed, thanks!